src/cbmc_starter_kit/template-for-repository/proofs/Makefile.common

```diff
@@ -711,15 +711,27 @@ endif
 # Early unwind all loops on DFCC mode; otherwise, only unwind loops in proof and project code
 $(HARNESS_GOTO)0500.goto: $(HARNESS_GOTO)0400.goto
 ifneq ($(strip $(USE_DYNAMIC_FRAMES)),)
+ifneq ($(strip $(APPLY_LOOP_CONTRACTS)),)
```

Comment:
This is a lot of lines of code for effectively two changes: adding `$(CBMC_DEFAULT_UNWIND)` and changing the description.

Would you consider a different approach: near the top of the file, do something like

```
ifneq ($(strip $(USE_DYNAMIC_FRAMES)),)
ifneq ($(strip $(APPLY_LOOP_CONTRACTS)),)
UNWIND_0500_FLAGS=$(CBMC_FLAG_UNWINDING_ASSERTIONS)
UNWIND_0500_DESC=
else
UNWIND_0500_FLAGS=$(CBMC_FLAG_UNWINDING_ASSERTIONS) $(CBMC_DEFAULT_UNWIND)
UNWIND_0500_DESC="$(PROOF_UID): unwinding all loops"
endif
```

then down here, you don't need to branch, just use the value of `UNWIND_0500_FLAG` in the command.

Reply:
I'm not going to block on this. It seems easier to to me to read the recipe this way, and if the reader cares about what the flags are they can check.

On the other hand it is less explicit and does not call attention to the fact that the flags might change depending on the proof.

Reply:
Great catch, Kareem! We should avoid code duplication is not less clear than how it was before.